### PR TITLE
Task 3 - Setup TDD (+ additional project setup)

### PR DESCRIPTION
**Pull request by oddestdan**

### CODE REVIEW

File: packages/server/src/services/achievement.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      );
+
+      Object.values(achievementsStatusMap).forEach((a) => {
+        expect(a.state).toEqual(StatusState.Pending);
+        expect(a.updated).toEqual(testDate);
+      });
+    });
+  });
+
+  describe('get', () => {
```
Comment (oleksandr-zhynzher) [inline packages/server/src/services/achievement.service.spec.ts]: better to name it `getActualAchievements`

File: packages/server/src/services/achievement.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { achievementsMock } from './__mocks__/achievement.mock';
+import { challengesMock } from './__mocks__/challenge.mock';
+import { calculateAchievementsStatus } from './index';
+import { StatusState } from '../enums';
+import { getActualAchievements } from './achievement.service';
+
+describe('achievement service', () => {
```
Comment (oleksandr-zhynzher) [inline packages/server/src/services/achievement.service.spec.ts:10]: Why you did`t test failure path

File: packages/server/src/services/challenge.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import { startNewChallenge } from './challenge.service';
+import { achievementsMock } from './__mocks__/achievement.mock';
+import { challengesMock } from './__mocks__/challenge.mock';
+import { tasksMock } from './__mocks__/task.mock';
+
+const mapToAchievementId = (achievement: Achievement): string => achievement.id;
+
+describe('challenge service', () => {
+  describe('startNewChallenge', () => {
+    let newChallenge: Challenge;
```
Comment (oleksandr-zhynzher) [inline packages/server/src/services/challenge.service.spec.ts:12]: it`s not good idea to create let and use it in all cases, lets move it to beforeEach

File: packages/server/src/services/challenge.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import { Achievement, Challenge } from '../models';
+import { startNewChallenge } from './challenge.service';
+import { achievementsMock } from './__mocks__/achievement.mock';
+import { challengesMock } from './__mocks__/challenge.mock';
+import { tasksMock } from './__mocks__/task.mock';
+
+const mapToAchievementId = (achievement: Achievement): string => achievement.id;
+
+describe('challenge service', () => {
+  describe('startNewChallenge', () => {
```
Comment (oleksandr-zhynzher) [inline packages/server/src/services/challenge.service.spec.ts:11]: Don't forgot to test corner cases, not only happy path
